### ISSUE TITLE
Features Request : auto tab-Completion 

### ISSUE THREAD

**Issue by JE-Amouel** [leo-arch/CliFM#313]
Hi, 

**Is your feature request related to a problem? Please describe.**
I'm really a fan of your idea of ​​being able to first call the name of a file then and then give the name of "the command", it really saves me time on a lot of things, seeing as when we're working between two The most annoying directory is having to type in entire directory names, so I often prefer to start with the most annoying. 

**Describe the solution you'd like**

I would like to be able to make the Tab completion automatic, 
ie I just want to be able to directly type Text and see the completion listing directly appear in our dear fzf, without need to press Tab each time, is that possible?.

**Describe alternatives you've considered**

THANKS

**Comment by leo-arch** [comment]
I think I've seen this somewhere. It would be like a mix between suggestions (which are automatic) and TAB completion. I'll take a look at it, but don't promise anything.

**Comment by JE-Amouel** [comment]
You're the best, I'm sure you'll succeed.

**Comment by JE-Amouel** [comment]
Clifm has already shown me lots of things that I thought were impossible in a terminal but you did it.

**Comment by JE-Amouel** [comment]
Hi, is there any update 👀

**Comment by leo-arch** [comment]
Hi @JE-Amouel. In order to implement this we want:

1. Trigger the completions window with any key press on the command line.
2. Update possible completions on the fly as command line input is processed.
3. Keep the focus on the command line and change to the completion window with some specific key press.

For the time being, and given the resources we handle, points 2 and 3 cannot be done. Once the completion window is triggered, we loose the focus until it is closed. Why? Because we spawn a new process each time the completions window is triggered.

In order for this to be achieved we need to implement a builtin and completely new completions system.

Summarizing, no, it's not possible (at least not now).

**Comment by JE-Amouel** [comment]
Thank you for your effort to make this possible on this program that I enjoy using daily, I would also like to tell you about my findings, which is a zsh Plugins, try it to see, maybe it can inspire you other ways. [zsh Autocomplete ](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://github.com/marlonrichert/zsh-autocomplete&ved=2ahUKEwjN1M20xtqKAxVT1gIHHd0cLE4QFnoECBUQAQ&usg=AOvVaw0GICppjnW89JJB22RhZ3Fp)

**Comment by leo-arch** [comment]
Yes, auto TAB completion is possible in `zsh` because it has its own builtin command line editor, including the completions system: `zle` (ZSH line editor).

And thanks for your kind words!

**Comment by JE-Amouel** [comment]
Do you think it would be possible to just add a Tab character after each character on the keyboard you type? And that without using a line editor? Each key on the keyboard would always be a character + Tab combination.

**Comment by leo-arch** [comment]
Yes, that's possible. But, since you'd lose the focus, which would be now on the completions window, you won't be able to keep typing. This is the point 1 I mentioned above: points 2 and 3 are the problematic ones.
